### PR TITLE
Externalise env modules into the nextflow config file

### PR DESCRIPTION
**Pull request by pditommaso**
Shows how externalise modules and how to create separate config profiles

### CODE REVIEW

**Comment by ewels** [comment]
Seriously? This works? Haha, that's awesome. _Much_ better than what we have. Thanks!

File: nextflow.config [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+profiles {
+  
+  module {
```
Comment (ewels) [inline nextflow.config:4]: And if we call this `standard` instead of `module`, then it will be used by default if `-profile` is not used?

Comment (pditommaso) [reply nextflow.config:4]: Yes.

Comment (pditommaso) [reply nextflow.config:4]: https://www.nextflow.io/docs/latest/config.html#config-profiles

**Comment by ewels** [comment]
Ok, this is great - we can move all of the stuff in `example_uppmax_config` into this file under a profile as well and users won't have to copy and paste the file as we're currently telling them to.

Thanks! 👍

**Comment by pditommaso** [comment]
Exactly, or [include](https://www.nextflow.io/docs/latest/config.html#config-include) that file into a profile

**Comment by ewels** [comment]
Even better - that will be more readable then. This is great, thanks!

**Comment by ewels** [comment]
Is it possible to set a default value for `-profile` in a user config file?

**Comment by pditommaso** [comment]
Do you mean a default profile name (that's `standard`) or default config attribute(s)?

**Comment by ewels** [comment]
So if I have a profile called `uppmax`, always use that by default instead of `standard` (so that I don't have to do `-profile uppmax` every time I run a pipeline).

I don't want to call the profile `standard` as then it'll be the default for everyone who runs the pipeline.

**Comment by pditommaso** [comment]
I see. The default one is `standard` and it's not possible to override it.

**Comment by ewels** [comment]
Ok, thanks. I'll just call it `standard` then and anyone running the pipeline elsewhere can override the settings with their `~/.nextflow/config` config file then I guess.

**Comment by pditommaso** [comment]
Um, they can override it providing a config file in the launching dir. The `~/.nextflow/config` won't work. The priority is the following (from lower to higher): 
- `~/.nextflow/config` 
- Pipeline config
- Launching config 
- Command line options

**Comment by ewels** [comment]
huh, ok that's not ideal. Is the Launching config one specified on the command line with `-c`?

I guess `-c ~/my_profile.config` is pretty much the same as `-profile my_profile`

**Comment by pditommaso** [comment]
Yes, you can also use a separate config with the `-c` option .

**Comment by ewels** [comment]
Final (?) question - is it possible to know what profile was used inside the nextflow script?

**Comment by pditommaso** [comment]
Yes, by using the `workflow.profile` implicit variable. See [here](https://www.nextflow.io/docs/latest/metadata.html#runtime-metadata).

**Comment by ewels** [comment]
Gotcha, thanks 👍  Apologies - did search the docs before asking but it didn't come up..

**Comment by pditommaso** [comment]
No pb. You are welcome.

**Pull request merged**